legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-h2/src/main/java/org/finos/legend/engine/persistence/components/relational/h2/H2Sink.java

```diff
@@ -172,4 +199,50 @@ public Optional<Optimizer> optimizerForCaseConversion(CaseConversion caseConvers
                 throw new IllegalArgumentException("Unrecognized case conversion: " + caseConversion);
         }
     }
+
+    public IngestorResult performBulkLoad(Datasets datasets, Executor<SqlGen, TabularData, SqlPlan> executor, SqlPlan ingestSqlPlan, Map<StatisticName, SqlPlan> statisticsSqlPlan, Map<String, String> placeHolderKeyValues)
+    {
+        executor.executePhysicalPlan(ingestSqlPlan, placeHolderKeyValues);
+
+        Map<StatisticName, Object> stats = new HashMap<>();
+
+        SqlPlan incomingRecordCountSqlPlan = statisticsSqlPlan.get(StatisticName.INCOMING_RECORD_COUNT);
+        if (incomingRecordCountSqlPlan != null)
+        {
+            long incomingRecordCount = (Long) executor.executePhysicalPlanAndGetResults(incomingRecordCountSqlPlan, placeHolderKeyValues)
+                .stream()
+                .findFirst()
+                .map(TabularData::getData)
+                .flatMap(t -> t.stream().findFirst())
+                .map(Map::values)
+                .flatMap(t -> t.stream().findFirst())
+                .orElseThrow(IllegalStateException::new);
+            stats.put(StatisticName.INCOMING_RECORD_COUNT, incomingRecordCount);
+        }
+
+        SqlPlan rowsInsertedSqlPlan = statisticsSqlPlan.get(StatisticName.ROWS_INSERTED);
+        if (rowsInsertedSqlPlan != null)
+        {
+            long rowsInserted = (Long) executor.executePhysicalPlanAndGetResults(rowsInsertedSqlPlan, placeHolderKeyValues)
+                .stream()
```

Comment:
This code seems duplicate of Lines 213-219.
Can we create a method that does this?

Reply:
Sure, changed